system76/pkgs/system76-io-dkms/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, fetchFromGitHub, fetchpatch, kernel }:
+let
+  version = "1.0.1";
+  sha256 = "0qkgkkjy1isv6ws6hrcal75dxjz98rpnvqbm7agdcc6yv0c17wwh";
+in
+stdenv.mkDerivation {
+  pname = "system76-io-dkms";
+  inherit version;
+
+  src = fetchFromGitHub {
+      owner = "pop-os";
+      repo = "system76-io-dkms";
+      rev = version;
+      inherit sha256;
+    };
+
+  hardeningDisable = [ "pic" ];
+  dontStrip = true;
+  dontPatchELF = true;
+
+  kernel = kernel.dev;
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  preBuild = ''
+    sed -e "s@/lib/modules/\$(.*)@${kernel.dev}/lib/modules/${kernel.modDirVersion}@" -i Makefile
+  '';
```

Comment:
This should do:

```
buildFlags = [
    "KERNELDIR=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
];
```

Reply:
Done (applied in the Nixpkgs PR).  I had to use `KERNEL_DIR` as that's what the Makefile uses.